test/python/http_recorder.py

```diff
@@ -0,0 +1,285 @@
+
+        Args:
+            ids2remove (dict): {request_path: [json_fields]}
+            cassette_dict (dict): a VCR cassette dictionary.
+        """
+
+        id_tracker = {}  # {old_id: new_id}
+        for path, fields in ids2remove.items():
+            responses = IdRemoverPersister.get_responses_with(path, cassette_dict)
+            for response in responses:
+                IdRemoverPersister.remove_ids_in_a_response(response, fields, path, id_tracker)
+        for old_id, new_id in id_tracker.items():
+            if not isinstance(old_id, str):
+                continue
+            for request in cassette_dict['requests']:
+                request.uri = request.uri.replace(old_id, new_id)
+
+    @staticmethod
+    def save_cassette(cassette_path, cassette_dict, serializer):
+        """
+        Extendeds FilesystemPersister.save_cassette. Replaces particular values (defined by
+        ids2remove) which are replaced by a dummy value. The full manipulation is in
+        cassette_dict, before saving it using FilesystemPersister.save_cassette
+
+        Args:
+            cassette_path (str): the file location where the cassette will be saved.
+            cassette_dict (dict): a VCR cassette dictionary. This is the information that will
+            be dump in cassette_path, using serializer.
+            serializer (callable): the serializer for dumping cassette_dict in cassette_path.
+        """
+        ids2remove = {'api/users/loginWithToken': ['id',
+                                                   'userId',
+                                                   'created'],
+                      'api/Jobs': ['id',
+                                   'userId',
+                                   'creationDate',
+                                   'qasms.executionId',
+                                   'qasms.result.date',
+                                   'qasms.result.data.time',
+                                   'qasms.result.data.additionalData.seed'],
+                      'api/Backends/ibmqx5/queue/status': ['lengthQueue'],
+                      'api/Backends/ibmqx4/queue/status': ['lengthQueue']}
+        IdRemoverPersister.remove_ids(ids2remove, cassette_dict)
+        super(IdRemoverPersister, IdRemoverPersister).save_cassette(cassette_path,
+                                                                    cassette_dict,
+                                                                    serializer)
+
+
+def purge_headers(headers):
+    """
+    Remove headers from the response.
+
+    Args:
+        headers (list): headers to remove from the response
+
+    Returns:
+        callable: for been used in before_record_response VCR constructor.
+    """
+
+    header_list = list()
+    for item in headers:
+        if not isinstance(item, tuple):
+            item = (item, None)
+        header_list.append((item[0], item[1]))
```

Comment:
Did you wanted to add a copy of the tuple? If not, you can simply add `ítem`.

Reply:
`header_list` is consumed as `(header, value) in header_list`. Some items are not tuples, I want them to have `None` as value.